secret-store container panics with --log-format-json=true

**What steps did you take and what happened:**
When running the helm chart v1.1.0 the container secrets-store in the daemonset panics with the following error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1632c4e]

goroutine 1 [running]:
go.uber.org/zap/zapcore.(*writerWrapper).Write(0xc0001d86d0, {0xc000152c00, 0x49d4527, 0x2abcee0})
	<autogenerated>:1 +0x2e
go.uber.org/zap/zapcore.(*ioCore).Write(0xc00038fec0, {0x0, {0xc07e0137f4ad9a7c, 0x49d4527, 0x2abcee0}, {0x0, 0x0}, {0x1a87aa3, 0x1c}, {0x1, ...}, ...}, ...)
	/go/pkg/mod/go.uber.org/zap@v1.19.1/zapcore/core.go:90 +0xb5
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00021bc80, {0xc000198380, 0x2, 0x2})
	/go/pkg/mod/go.uber.org/zap@v1.19.1/zapcore/entry.go:220 +0x1d9
github.com/go-logr/zapr.(*zapLogger).Info(0xc0003fc000, 0x0, {0x1a87aa3, 0x7af3f3}, {0xc000252840, 0x2, 0x2})
	/go/pkg/mod/github.com/go-logr/zapr@v1.2.0/zapr.go:198 +0xa5
github.com/go-logr/logr.Logger.Info({{0x1cee820, 0xc0003fc000}, 0x0}, {0x1a87aa3, 0x1c}, {0xc000252840, 0x2, 0x2})
	/go/pkg/mod/github.com/go-logr/logr@v1.2.0/logr.go:249 +0xd0
k8s.io/klog/v2.(*loggingT).infoS(0x7b4546, 0xc00067f830, {0x0, 0x0}, 0x19a0840, {0x1a87aa3, 0xc00067f840}, {0xc000252840, 0x2, 0x2})
	/go/pkg/mod/k8s.io/klog/v2@v2.30.0/klog.go:795 +0x113
k8s.io/klog/v2.InfoS(...)
	/go/pkg/mod/k8s.io/klog/v2@v2.30.0/klog.go:1438
sigs.k8s.io/secrets-store-csi-driver/pkg/metrics.InitMetricsExporter()
	/go/src/sigs.k8s.io/secrets-store-csi-driver/pkg/metrics/exporter.go:35 +0xcc
main.main()
	/go/src/sigs.k8s.io/secrets-store-csi-driver/cmd/secrets-store-csi-driver/main.go:104 +0x19e
```

**What did you expect to happen:**

Container should start without problems.

**Anything else you would like to add:**
Without the `--log-format-json=true` flag the container starts without problem.


**Which provider are you using:**
HashiCorp Vault

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
  - Helm Chart v1.1.0
  - k8s.gcr.io/csi-secrets-store/driver:v1.1.0
- Kubernetes version: (use `kubectl version`): v1.21.6+vmware.1

--- a/cmd/secrets-store-csi-driver/main.go
+++ b/cmd/secrets-store-csi-driver/main.go
@@ -43,6 +43,7 @@ import (
 	"k8s.io/client-go/kubernetes"
 	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
+	"k8s.io/component-base/config"
 	json "k8s.io/component-base/logs/json"
 	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
@@ -89,7 +90,8 @@ func main() {
 	flag.Parse()
 
 	if *logFormatJSON {
-		logger, _ := json.NewJSONLogger(nil, nil)
+		jsonFactory := json.Factory{}
+		logger, _ := jsonFactory.Create(config.FormatOptions{})
 		klog.SetLogger(logger)
 	}
 	if *enableProfile {

